<launch>

    <arg name="is_sim" value="0" />
    <arg name="real_time" value="0"/>
    <arg name="target_timestep" value="1.0"/>

    <rosparam file="$(find gtddp_drone)/launch/constants/pursuit_real_params.yaml" command="load" />

    <!-- Debugging -->
    <param name="debug_mode" value="true" />

    <!-- GT-DDP nodes -->
    <node name="pursuit_control_node" pkg="gtddp_drone" type="pursuit_control_node" output="screen">
        <param name="is_simulation" value="$(arg is_sim)"/>
    </node>
    <node name="gtddp_optimize_node" pkg="gtddp_drone" type="gtddp_optimize_node" output="screen">
        <param name="is_simulation" value="$(arg is_sim)"/>
        <param name="is_real_time" value="$(arg real_time)"/>
    </node>

    <!-- PURSUER -->
    <group ns="drone1">
        <!-- AR.Drone Driver -->
        <include file="$(find ardrone_autonomy)/launch/ardrone.launch">
            <!-- Change the ip argument to configure the IP address of the AR.Drone to connect to -->
            <arg name="ip" value="192.168.1.11"/>
            <!-- CHANGE THE ABOVE LINE TO BE CORRECT -->
        </include>
    </group>

    <!-- EVADER -->
    <group ns="drone2">
        <!-- AR.Drone Driver -->
        <include file="$(find ardrone_autonomy)/launch/ardrone.launch">
            <!-- Change the ip argument to configure the IP address of the AR.Drone to connect to -->
            <arg name="ip" value="192.168.1.12"/>
            <!-- CHANGE THE ABOVE LINE TO BE CORRECT -->
        </include>
    </group>

    <!-- VICON -->
        <!-- Make sure to follow compile instructions in mocap_vicon -->
        <node pkg="mocap_vicon" type="mocap_vicon_node" name="vicon" >
            <!-- Set the server address to the IP of the computer running the VICON Tracker program -->
            <!-- Set the port number according to the UDP port set in Tracker -->
            <!-- Format: IP:PORT like 192.168.X.ABC:PORT -->
            <param name="server_address" value="192.168.1.119:801"/>
            <param name="frame_rate" value="100"/>
            <param name="max_accel" value="10.0"/>
            <param name="publish_tf" value="false"/>
            <param name="fixed_frame_id" value="mocap"/>
            <rosparam param="model_list">[]</rosparam>
            <!--remap from="vicon/model_name/odom" to="/model_name/odom"/-->
        </node>

    <!-- Keyboard node -->
    <node pkg="gtddp_drone" type="keyboard_node.py" name="keyboard_node" output="screen"/>

</launch>
